source/NVDAObjects/IAccessible/__init__.py

```diff
@@ -1300,7 +1300,7 @@ def event_selectionWithIn(self):
 
 	def _get_isPresentableFocusAncestor(self):
 		IARole = self.IAccessibleRole
-		if IARole == oleacc.ROLE_SYSTEM_CLIENT and self.windowStyle & winUser.WS_SYSMENU:
+		if IARole == oleacc.ROLE_SYSTEM_CLIENT and self.event_objectID==winUser.OBJID_CLIENT and self.event_childID==0 and self.windowStyle & winUser.WS_SYSMENU:

```

Comment:
I think we should have a comment in this block explaining this, as it's a bit obscure. Something like this:

```
# This is the root object of a top level window.
# #4300: We check the object and child ids as well because there can be "clients" other than the root.
```